doc/man/8/rbdmap-generator.rst

```diff
@@ -0,0 +1,153 @@
+   supported.
+
+#. image-or-snap-spec is the image or snapshot specification, in the form of
+   [<pool-name>/[<namespace>/]]<image-name>[@<snap-name>]. If only <image-name>
+   is specified, <pool-name> defaults to "rbd".
+
+#. generic-options is a comma-delimited list of options. This field is optional.
+   The supported options are listed below.
+
+#. map-options is passed unchanged to ``--options`` argument of ``rbd device
+   map`` command. This field is optional.
+
+#. unmap-options is passed unchanged to ``--options`` argument of ``rbd device
+   unmap`` command. This field is optional.
+
+If successful, the generated unit maps the image to a ``/dev/rbdX`` device, at
+which point a udev rule is triggered to create a friendly device name symlink,
+``/dev/rbd/<image-or-snap-spec>``, pointing to the mapped device.
+
+When writing ``/etc/fstab`` entries for RBD images, the ``_netdev`` mount option
+should be used. Otherwise, a dependency loop might be created where the mount
+point will be pulled in by local-fs.target, while the service to configure the
+network is usually only started after the local file system has been mounted.
+
+For capability reason, if /etc/ceph/rbdtab file is not present, /etc/ceph/rbdmap
+file will be read and parsed in a way compatible with the old ``rbdmap`` script.
+
+Supported Generic Options in Config File
+----------------------------------------
+
+.. option:: noauto, auto
+
+   With ``noauto``, the device will not be added as a dependency for
+   rbdmap.target. This means that it will not be automatically mapped on boot,
+   unless something else pulls it in. In particular, if the device is used for a
+   mount point, it'll be mapped automatically during boot, unless the mount
+   point itself is also disabled with ``noauto``.
+
+.. option:: nofail, fail
+
+   With ``nofail``, this device will not be a hard dependency of rbdmap.target.
+   It'll still be pulled in and started, but the system will not wait for the
+   device to show up, and boot will not fail if this is unsuccessful. Note that
+   other units that depend on the RBD device may still fail. In particular, if
+   the device is used for a mount point, the mount point itself also needs to
+   have the ``nofail`` option, or the boot will fail if the device is not mapped
+   successfully.
+
+.. option:: x-systemd.requires=, x-systemd.requires-mounts-for=, x-systemd.before=, \
+   x-systemd.after=, x-systemd.wanted-by=, x-systemd.required-by=
+
+   See systemd.unit(5) for details.
+
+All other options that not start with "x-" are passed to ``rbd device map``
+command directly as command-line arguments.. See :program:`rbd`\(8) for details
+
+
+Examples
+========
+
+Example ``/etc/ceph/rbdtab`` for three RBD images: "bar1" in pool "rbd"; "bar2"
+and "bar3" in pool "foopool"::
+
+    krbd bar1
+    krbd foopool/bar2    id=admin,noauto
+    krbd foopool/bar3    id=admin        lock_on_read,queue_depth=1024  force
```

Comment:
Good point. I think maybe we can switch to a more complex config format? e.g. INI style like [systemd.netdev(5)](https://www.freedesktop.org/software/systemd/man/systemd.netdev.html)

`/etc/ceph/rbd/foopool-bar2.rbd`
```ini
[RBD]
Spec=foopool/bar2
Auto=no
Id=admin

[Mount]
LockOnRead=yes
QueueDepth=1024

[Umount]
force=yes
```

Reply:
crypttab uses "none" or "-" as a placeholder for its third field, we can also follow it. So that we can write `krbd foopool/bar3 none lock_on_read`. What do you think?

Reply:
LGTM